consensus/istanbul/backend/announce.go

```diff
@@ -34,14 +39,15 @@ import (
 // define the istanbul announce message
 
 type announceMessage struct {
-	Address   common.Address
-	EnodeURL  string
-	View      *istanbul.View
-	Signature []byte
+	Address            common.Address
+	IncompleteEnodeURL string //TODO(nguo) remove this field
```

Comment:
IIUC, the incompleteEnodeURL contains the node ID (e.g. something like this:  `enode://352352235235235...@`.

One suggestion is to make the field more structured in that you replace it with the nodeID.